File: finstack/valuations/tests/golden/runners/calibration_swaption_vol.rs
//! Domain runner for swaption volatility calibration golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Swaption volatility calibration golden runner.
pub struct CalibrationSwaptionVolRunner;

impl DomainRunner for CalibrationSwaptionVolRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::calibration_common::run_sabr_cube_fixture(fixture)
    }
}

File: finstack-py/tests/golden/runners/calibration_swaption_vol.py
"""Domain runner for swaption volatility calibration golden fixtures."""

from __future__ import annotations

from tests.golden.runners.calibration_common import run_sabr_cube_fixture
from tests.golden.schema import GoldenFixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Require executable swaption-vol calibration inputs before running."""
    return run_sabr_cube_fixture(fixture)
